tests/components/mqtt/test_init.py

```diff
@@ -255,6 +255,23 @@ async def test_service_call_without_topic_does_not_publish(
     assert not mqtt_mock.async_publish.called
 
 
+async def test_service_call_mqtt_entry_does_not_publish(
+    hass: HomeAssistant, mqtt_client_mock: MqttMockPahoClient
+) -> None:
+    """Test the service call if topic is missing."""
+    assert await async_setup_component(hass, mqtt.DOMAIN, {})
+    with pytest.raises(ServiceValidationError):
```

Comment:
Should we add the string validation in here?
```suggestion
    with pytest.raises(ServiceValidationError, matches="..."):
```

Reply:
May be